functions.php

```diff
@@ -53,6 +53,22 @@ function as_schedule_single_action( $timestamp, $hook, $args = array(), $group =
  * @return int The action ID.
  */
 function as_schedule_recurring_action( $timestamp, $interval_in_seconds, $hook, $args = array(), $group = '', $unique = false ) {
+	$interval = (int) $interval_in_seconds;
+
+	// We expect an integer and allow it to be passed using float and string types, but otherwise
+	// should reject unexpected values.
+	if ( ! is_numeric( $interval_in_seconds ) || $interval_in_seconds != $interval ) {
+		_doing_it_wrong(
+			__METHOD__,
+			sprintf( 
+				/* translators: 1: provided value 2: provided type. */
+				__( 'An integer was expected but "%1$s" (%2%s) was received.', 'action-scheduler' ),
+				$interval_in_seconds,
+				gettype( $interval_in_seconds )
+			),
+			ActionScheduler_Versions::instance()->latest_version()
+		);
+	}
 	if ( ! ActionScheduler::is_initialized( __FUNCTION__ ) ) {
 		return 0;
 	}
```

Comment:
I propose we keep this block at the top of the method.

That is, if Action Scheduler is not initialized, then bail out early *(before* testing if `$interval_in_seconds` is valid). It occurs that this may be safer since we are making use of `ActionScheduler_Versions::instance()` in our `_doing_it_wrong()` call.